### PR TITLE
Submission of ntbox and bamm 

### PR DESCRIPTION
**Pull request by luismurao**
`ntbox` functions

occ_acquisition: 
    - *ntbox::search_gbif_data*. Search GBIF data using the function occ from  the spocc package and show a leaflet map of occurrence locations
occ_cleaning: 
   - *ntbox::clean_dup*. Cleans duplicated longitude and latitude data using threshold distance which is the distance between points to be considered duplicates.
   - GUI functions to curate occurrence data. 
data_integration: 
    - *ntbox::swd_format*. Prepares data using Maxent's samples with data (SWD) format. 
    - *ntbox::sample_envbg*. Generate random environmental background data
env_collinearity: 
    - *ntbox::correlation_finder*. The function finds out which 
variables have strong correlations according to a correlation threshold. The output returns a list of variable names that can summarize the information and removes redundant variables.
env_process: 
   - *ntbox::spca*. Performs PCA transformation of environmental layers.
   - *ntbox::get_bio_oracle*. Get environmental layers from Bio-Oracle. 
   - *ntbox::get_chelsa*. Get bioclimatic layers from CHELSA in current and future scenarios 
bias: 
   - *ntbox::biaslayer*. Function to create a bias layer
study_region: Available in GUI Trace polygon of the study region or upload it
as a shapefile
backg_sample: 
   - *ntbox::sample_envbg*. Generate random environmental background data
data_partitioning: Available in GUI Simple random partition of occurrence data
mod_fit: 
   - *ntbox::cov_center*. Function to compute the minimum volume covariance matrix 
   of an ellipsoid niche model. The function also computes ellipsoid volume semi-axes length and coordinates. 
   - *ntbox::maxent_call*. Fit maxent models from R
mod_tuning: 
    - *ntbox::ellipsoid_selection*. Performs model calibration, evaluation, and selection for ellipsoid models. The function fits ellipsoid models for all 
possible combinations of n variables taken by m. Each model is evaluated using 
the partial ROC test, omission rate criteria, and AUC
mod_evaluate: 
    - *ntbox::pROC*. Performs the partial ROC test and computes AUC values.
    - *ntbox::omission_rate*. Computes the omission rate of a model.  
    - *ntbox::inEllipsoid*. Evaluation in E-space. Determines whether a point is 
    inside or outside of an ellipsoid. 
    - *ntbox::ellipsoid_omr*. Computes the omission rate of ellipsoid models.
    - *ntbox::binomial_test*. Performa the binomial test
pred_extrapolation: 
    - *ntbox::mop*. Performs Mobility-Oriented Parity analysis. 
    - *ntbox::ntb_mess*. Estimates Multivariate Environmental Similarity Surfaces (MESS).
    - *ntbox::exdet_multvar*. Estimates Exdet multivariate metric: NT2 metric. 
    - *ntbox::exdet_univar*. Estimates Exdet univariate metric: NT1 metric
pred_inspect: Available in GUI. Visualizes ellipsoid models in E-space, 
shows model projection in G-space, response curves for ellipsoid models and 
bioclim models
post_processing: 
   - *ntbox::bin_model*. Binarizes a model using a threshold. 
   - Available in GUI functions to crop and mask data using R as a SIG.
gui: 
   - *ntbox::run_ntbox*. Opens the GUI of ntbox.
   
`bamm` functions

mod_stack: 
    - *bamm::models2pam*. Function to convert binary raster models to a Presence Absences Matrix.
    - *bamm::csim2pam*. Converts community simulation object into a Presence Absence Matrices (PAM) for a given simulation steps.
    - *bamm::pam2richness*. Converts Presence Absence Matrix (pam object) to richness raster.
mod_multispecies:
    - *bamm::community_sim*. Estimate community dynamics using the bamm framework
    - *bamm::diversity_range_analysis*. Diversity range analysis and estimation of 
    biodiversity indices (alpha, dispersion field, gamma) related to diversity-range plots.
pred_general: 
    - *bamm::predict*. Predicts dispersal dynamics under suitability change scenarios
pred_inspect:
    - *bamm::plot*. Plot method for objects of class diversity_range bamm.
    - *bamm::bam_clusters*. Function to estimate the connectivity of suitable areas 
    given an adjacency matrix (defined using a dispersal capacity hypothesis). 
    The output contains 1) connections: a data.frame with three columns where the first and the second represent (x and y) centroid coordinates of the niche model and the third column with the cluster-ID where they belong. 2) interactive_map: a leaflet map of connected suitable pixels shown in the same color. 3) A RasterLayer of connected suitable pixels.
    - *bam::csd_plot*. Gives an estimate of the number of geographic clusters given a set of dispersal hypotheses and a suitability raster
post_processing: 
    - *bammm::sdm_sim*. Simulate single-species dispersal dynamics using the BAM
    framework.
    - *bamm::jaccard*. Estimates the Jaccard index for comparing two binary predictions
    - *bamm::pam2bioindex*. Estimates various biodiversity indices for a certain 
    presence-absence matrix.

### CODE REVIEW

**Comment by sylvainschmitt** [comment]
@luismurao , thanks for your submission. Can you fix the following [issue](https://github.com/sylvainschmitt/sdmverse/actions/runs/4577966431/jobs/8112163350?pr=16):

> Error in sdmverse::check_cran_metadata() : 
-For package bamm, version is defined as '2.0.4' but should be defined as '0.4.3' according to CRAN.

**Comment by sylvainschmitt** [comment]
I'm looking for reviewers: @EduardoArle , @gepinillab , @hannahlowens , @jamiemkass , @sgvignali, @codelab3 , @adamlilith; @MayaGueguen  are you available? Please have a look at the [reviewing guidelines](https://github.com/sylvainschmitt/sdmverse/blob/main/REVIEWING.md)

**Review by sylvainschmitt — APPROVED** [review]
Thanks for the nice description of your two packages in the submission. At the exception of bamm version, everything looks good to me. I'll let time for somebody else to review, but if anybody react I'll accept your PR. After merging your pull request, you will be added to the repository contributors. Please expect to be contacted for a review request in the future.

**Comment by hannahlowens** [comment]
I can review.

**Comment by sylvainschmitt** [comment]
> I can review.

Thanks, please do.

File: inst/extdata/packages/ntbox.yaml
Comment (hannahlowens) [inline inst/extdata/packages/ntbox.yaml:13]: This is an aside from the `sdmverse` review, but you may want to note somehow in the documentation that it is not recommended to use occurrence data directly from `spocc` in studies that are intended for publication, since GBIF datasets should be cited using a DOI. See [link](https://www.gbif.org/citation-guidelines) for more information.

File: inst/extdata/packages/ntbox.yaml
Comment (hannahlowens) [inline inst/extdata/packages/ntbox.yaml:29]: No explanation provided for `ntbox` pred_general in submission comments.

File: inst/extdata/packages/ntbox.yaml
Comment (hannahlowens) [inline inst/extdata/packages/ntbox.yaml:34]: No explanation provided for `ntbox` metadata in submission comments.

File: inst/extdata/packages/bamm.yaml
Comment (hannahlowens) [inline inst/extdata/packages/bamm.yaml]:
```suggestion
mod_multispecies: no
```
In my understanding from the explanation, it seems to me more that `bamm` is doing post-processing analysis on distributional models for several species. I think `mod_multispecies` refers more to whether the package itself *generates* models for multiple species. @sylvainschmitt, is that what you meant?

**Review by hannahlowens — CHANGES_REQUESTED** [review]
Please provide information on the two fields requested.

File: inst/extdata/packages/ntbox.yaml
Comment (luismurao) [reply inst/extdata/packages/ntbox.yaml:29]: Thanks @hannahlowens, please check this out:
pred_general:
    - *ntbox::ellipsoid_fit*. Makes prediction of ellipsoid models in G-space.
    - Functions to make predictions of bioclim models available in GUI.

File: inst/extdata/packages/ntbox.yaml
Comment (luismurao) [reply inst/extdata/packages/ntbox.yaml:13]: Thank you for this recommendation. I just added a note regarding this topic  in the documentation of the function. Please see
https://github.com/luismurao/ntbox/blob/ba483fd35924367a272582a8789aea8ea9803364/R/search_gbif_data.R#L11

File: inst/extdata/packages/ntbox.yaml
Comment (luismurao) [reply inst/extdata/packages/ntbox.yaml:34]: metadata: 
    - Available in GUI. A summary of the workflow is produced for use as 
    a stand-alone algorithm or as part of research reports.

File: inst/extdata/packages/ntbox.yaml
Comment (sylvainschmitt) [reply inst/extdata/packages/ntbox.yaml:29]: @hannahlowens , I'll let you resolve the comment or ask for more.

File: inst/extdata/packages/ntbox.yaml
Comment (sylvainschmitt) [reply inst/extdata/packages/ntbox.yaml:34]: The summary of all the code used in the GUI for the workflow is not metadata for me. The aim of sdmverse is to disciminate packages about their core functionality. It does not seems to me that this is the case for metadata in ntbox. @hannahlowens what do you think?

Comment (sylvainschmitt) [reply inst/extdata/packages/ntbox.yaml:34]: PS: And thanks for the review, the fact that I missed it and that you spotted it means that I should not review alone a package.

File: inst/extdata/packages/bamm.yaml
Comment (sylvainschmitt) [reply inst/extdata/packages/bamm.yaml]: To summarize, what we wrote with @jamiemkass & @adamlilith is:

> mod_multispecies: Function(s) that model multiple species at once to estimate community composition / diversity (i.e., not a “stacked model; yes/no)

And what @luismurao described is:

> - bamm::community_sim. Estimate community dynamics using the bamm framework
> - bamm::diversity_range_analysis. Diversity range analysis and estimation of biodiversity indices (alpha, dispersion field, gamma) related to diversity-range plots.

I feel this is more about post-processing too, but I'm a bit unsure. I should not decide alone the rules. So @luismurao you can explain further why you think it should be included in multispecies modelling. And @hannahlowens , @jamiemkass  and @adamlilith I'm interested by what your minds on the question.

**Comment by sylvainschmitt** [comment]
@luismurao , congrats your metadata passed all checks. We just have to resolve the three opened comments and your packages can be included.

File: inst/extdata/packages/ntbox.yaml
Comment (hannahlowens) [reply inst/extdata/packages/ntbox.yaml:34]: For me, it depends a bit on the nature of the summary--does it contain information on the data that went in, and/or on the processes performed? I referred back to the other three packages that claim metadata features. `occCite` specifically collects and curates occurrence metadata, and generates citation lists. `enmEval` and `wallace` behave a bit more like what Luis is describing, so for the sake of consistency, I'd say we either allow metadata as a feature for `ntbox` or remove it from `enmEval` and `wallace`.

File: inst/extdata/packages/ntbox.yaml
Comment (hannahlowens) [reply inst/extdata/packages/ntbox.yaml:29]: Looks good.

File: inst/extdata/packages/ntbox.yaml
Comment (jamiemkass) [reply inst/extdata/packages/ntbox.yaml:34]: Just for the record, `ENMeval` and `wallace` produce RMMS metadata. `wallace` also produces an RMD file that reproduces the analysis in R, which is admittedly not really metadata by itself. However, looking at 10) in the [`ntbox` vignette](https://luismurao.github.io/ntbox/articles/gui_reference.html#10-saving-your-analysis), this is more like metadata material in my opinion -- graphics and stats showing specifics about the analysis (not just the results). Any other opinions?

File: inst/extdata/packages/bamm.yaml
Comment (jamiemkass) [reply inst/extdata/packages/bamm.yaml]: I guess it depends on whether or not `bamm` makes the models that generate the predictions which are used to estimate diversity indices or not. If the answer is no, I would not call it a "multispecies SDM" because it is using SDM results (post-processing, as Sylvain said).

**Comment by sylvainschmitt** [comment]
@luismurao , your contribution can be integrated without further edits. After merging your pull request, you will be added to the repository contributors. Please expect to be contacted for a review request in the future.

**Pull request merged**
